lib/view_component/slot_v2.rb

```diff
@@ -58,6 +62,10 @@ def method_missing(symbol, *args, &block)
       @_component_instance.public_send(symbol, *args, &block)
     end
 
+    def html_safe?
+      to_s.html_safe
```

Comment:
```suggestion
      to_s.html_safe?
```

Reply:
at some point I have to assume the HTML is safe, otherwise `html_safe?` will always be `false`.
What I can do is change the `to_s` capture to:

```rb
@content = view_context.capture do
        if defined?(@_component_instance)
          # render_in is faster than `parent.render`
          @_component_instance.render_in(view_context, &@_content_block)
        elsif defined?(@_content)
          @_content
        elsif defined?(@_content_block)
          @_content_block.call
        end
      end.html_safe
```

and here use `to_s.html_safe?`, but this is basically the same as defining 

```rb
def html_safe?
  true
```

since we are always assuming safety :/

Reply:
> at some point I have to assume the HTML is safe, otherwise `html_safe?` will always be `false`.

I thought that slim/erb/haml would tell us if it's `html_safe` or not by us calling `to_s.html_safe?`? I don't think we'd have to mark it as safe anywhere explicitly in the library.

Would it be worth adding a test that outputs content we'd want to ensure is escaped?

 e.g. something roughly like

```ruby
class UserBioComponent < ViewComponent::Base
  renders_one :header, -> (user:) do
    user.bio # This is unsafe, since this could be something like `<script>alert('hello')</script>`
  end
end

# erb <%= header %>

# test
def test_slot_escapes_unsafe_content
  user = OpenStruct.new(bio: '<script>alert('hello')</script>')
  render_inline UserBioComponent.new { |component| component.header(user: user) }

  assert_text '&lt;script&gt;alert('hello')&lt;/script&gt;'
end
```

Reply:
from what I see here, Rails always sets strings as unsafe no matter what. e.g:
```rb
"foo".html_safe? -> false
"<div>hello</div>".html_safe? -> false
```

If we want to escape unsafe parts we'd need something like [sanitize](https://github.com/rgrove/sanitize).

Reply:
What I actually don't know is if the content safety is indeed a responsibility of the gem or of the user.
I see the `content` use something like React's [dangerouslySetInnerHTML](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml) which infers that the user knows what is being rendered

Reply:
> from what I see here, Rails always sets strings as unsafe no matter what. e.g:

That is true, but in this case `@content` should be the result of the slim template rendering, so I'd expect it to be marked as `html_safe?` appropriately by slim itself.

> What I actually don't know is if the content safety is indeed a responsibility of the gem or of the user.

In the example above, I'd say it's on the framework (us) to ensure the consumer of the API can't easily create a vulnerability.

Similar to how Rails works, I'd expect a user to have to opt-in to rendering the user-input as-is with the `raw` helper method.